src/modlunky2/ui/install.py

```diff
@@ -283,6 +288,14 @@ def download_file(call, url: str, dest_path: Path):
     with dest_path.open("wb") as dest_file:
         shutil.copyfileobj(contents, dest_file)
 
+    if dest_path.suffix == ".lua":
+        lua_path = Path(dest_path)
+
+        try:
+            lua_path.rename(lua_path.parent / "main.lua")
```

Comment:
Rather than saving and renaming you could probably just change the `dest_path` before saving the file. but I think this logic might be better moved up to the `download_mod_file` function as this function is meant to be more of a generic download.

in download_mod_file you could check if the `filename` ends in `.lua` and if pass `pack_dir/"main.lua"`